text/0000-dynamic-import-hints.md

```diff
@@ -0,0 +1,257 @@
+* Cannot combine multiple values using operators(i.e `"KEY": "LARGE"&"SMALL"` or `"KEY": "LARGE && SMALL"` will not qualify as a valid hint)
+
+##### Mutiple Hints:
+When multiple hint comments are specifed for a single dynamic import statement, only the fist hint is considered.
+Example:
+```js
+import("c-bar-mobile"
+    /* "@salesforce/client/formFactor": "SMALL"*/
+    /* "@salesforce/userPermission/ViewSetup": "true"*/
+);
+```
+In the above example, only the `formFactor` hint is considered.
+
+#### Semantics
+The [_ModuleSpecifier_](http://www.ecma-international.org/ecma-262/6.0/index.html#sec-imports) of the import statement should be a string literal. The compiler cannot analyze data about a module name that will be discovered or learned at runtime. For example, in the following case no hint metadata will be gathered:
+```js
+const bar = '/modules/my-module.js';
+function foo() {
+    import(
+        bar
+        /* "@salesforce/client/formFactor": "SMALL" */
+    ).then(module => {
+        success();
+    });
+}
+```
+If a hint does not follow the above grammar, metadata gathering for that specific hint will be skipped. However, if there are other hints in the same module that follow the grammar, metadata will be gathered.
+
+#### Examples
+Valid
+```js
+// Single hint
+import("c-bar-mobile"/* "@salesforce/client/formFactor": "SMALL"*/);
+import("c-bar-mobile"/* "KEY.SUBKEY": "VALUE"*/);
+import("c-bar-mobile"/* "KEY/SUBKEY1": "VALUE"*/);
+import("c-bar-mobile"/* "KEY": "true"*/);
+import("c-bar-mobile"/* "KEY": "1"*/);
+```
+Invalid
+```js
+import("c-bar-mobile"/* "KEY SUBKEY": "SMALL"*/);
+import("c-bar-mobile"/* "😀": "1"*/);
+import("c-bar-mobile"/* @salesforce/client/formFactor = SMALL*/);
+
+import("c-bar-mobile" /* "@salesforce/client/formFactor": "SMALL && LARGE"*/);
+import("c-bar-mobile" /* "@salesforce/client/formFactor": "SMALL" && "@salesforce/client/formFactor": "LARGE"*/);
+
+// Dynamic module name specifier
+const bar = 'module-name';
+import(bar/* "@salesforce/client/formFactor": "SMALL" */)
+```
+
+#### Separation of Concern
+Dynamic imports hint metadata gathering happens during component compilation. An invariant of this process is that the metadata collection will **not** fail due to a bad hint. This could be due to a hint not following specified grammar or not using a well understood hint value.
+
+Additional validation and post processing of the gathered metadata is out of the scope of this RFC. That logic will be the responsibility of the application framework that consumes the metadata.
+
+Similarly, any design time validations of the hint grammar should be handled by linting code. For example, on salesforce platform, allowed hint keys might be restricted to only `@salesforce` scoped specifiers. The hint values will be validated based on the relevant value for the specifier. An ESLint rule will enforce this restriction via the [eslint-plugin-lwc](https://github.com/salesforce/eslint-plugin-lwc).
```

Comment:
> Similarly, any design time validations of the hint grammar should be handled by linting code.

I don't think the term design time makes much sense here. Validation is done by the consumer of the compiled output. Linting might also be used as a way to validate import hints without having to compile a module.